packages/flutter_tools/lib/src/test/runner.dart

```diff
@@ -20,141 +20,177 @@ import 'flutter_web_platform.dart';
 import 'test_wrapper.dart';
 import 'watcher.dart';
 
-/// Runs tests using package:test and the Flutter engine.
-Future<int> runTests(
-  TestWrapper testWrapper,
-  List<String> testFiles, {
-  Directory workDir,
-  List<String> names = const <String>[],
-  List<String> plainNames = const <String>[],
-  bool enableObservatory = false,
-  bool startPaused = false,
-  bool disableServiceAuthCodes = false,
-  bool ipv6 = false,
-  bool machine = false,
-  String precompiledDillPath,
-  Map<String, String> precompiledDillFiles,
-  @required BuildMode buildMode,
-  bool trackWidgetCreation = false,
-  bool updateGoldens = false,
-  TestWatcher watcher,
-  @required int concurrency,
-  bool buildTestAssets = false,
-  FlutterProject flutterProject,
-  String icudtlPath,
-  Directory coverageDirectory,
-  bool web = false,
-  String randomSeed = '0',
-}) async {
-  // Configure package:test to use the Flutter engine for child processes.
-  final String shellPath = globals.artifacts.getArtifactPath(Artifact.flutterTester);
-  if (!globals.processManager.canRun(shellPath)) {
-    throwToolExit('Cannot execute Flutter tester at $shellPath');
-  }
+/// A class that abstracts launching the test process from the test runner.
+abstract class FlutterTestRunner {
+  const factory FlutterTestRunner() = _FlutterTestRunnerImpl;
 
-  // Compute the command-line arguments for package:test.
-  final List<String> testArgs = <String>[
-    if (!globals.terminal.supportsColor)
-      '--no-color',
-    if (startPaused)
-      '--pause-after-load',
-    if (machine)
-      ...<String>['-r', 'json']
-    else
-      ...<String>['-r', 'compact'],
-    '--concurrency=$concurrency',
-    for (final String name in names)
-      ...<String>['--name', name],
-    for (final String plainName in plainNames)
-      ...<String>['--plain-name', plainName],
-    '--test-randomize-ordering-seed=$randomSeed',
-  ];
-  if (web) {
-    final String tempBuildDir = globals.fs.systemTempDirectory
-      .createTempSync('flutter_test.')
-      .absolute
-      .uri
-      .toFilePath();
-    final bool result = await webCompilationProxy.initialize(
-      projectDirectory: flutterProject.directory,
-      testOutputDir: tempBuildDir,
-      testFiles: testFiles,
-      projectName: flutterProject.manifest.appName,
-      initializePlatform: true,
-    );
-    if (!result) {
-      throwToolExit('Failed to compile tests');
+  /// Runs tests using package:test and the Flutter engine.
+  Future<int> runTests(
+    TestWrapper testWrapper,
```

Comment:
This method has a lot of parameters. Should we have a TestRunnerConfig class to encapsulate and document them?

Reply:
Probably - I'd rather do that as a separate effort to avoid breaking things (right now I'm just moving it to a class so we can inject and test this)